### DISCUSSION TITLE
I designed some 3D Printable Cases for the RP6502

### DISCUSSION THREAD

**Discussion by vruumllc** [picocomputer/community#50] [Q&A]
Here are three 3D printable case designs I made for the RP6502:

1. The rp6502 directory files document a case just for the RP6502 board alone. 
2.  The rp6502_hub directory documents a case for the RP6502 board and a small internal USB hub.
3. The rp6502_hub_video_audio documents a case for the RP6502 board with internal USB hub, VGA display, audio amp, and speakers.

For each design you can either have a solid top or a see-through top. You can reuse the official acrylic 'case' for the window.

https://github.com/vruumllc/RP6502-Cases

![rp6502_front](https://github.com/picocomputer/community/assets/145504135/5880bb42-6624-446d-ac27-79601f7bf9b0)
![rp6502_hub_front](https://github.com/picocomputer/community/assets/145504135/5fe51096-152f-4260-b09c-15afb74a197d)
![rp6502_hub_video_audio_front](https://github.com/picocomputer/community/assets/145504135/828acba1-9d27-4462-8c69-0f95e3ccfd3e)

I built the rp6502_hub_video_audio. Check it out in the 'First Builds' thread.

**Comment by rumbledethumps** [comment]
I ordered the hub and cable. Have everything else for the hub-case.

**Reply by vruumllc** [reply]
Cool! I'm thrilled you are going to build one.

**Comment by vruumllc** [comment]
Better colors now. As we used to say back in my mountain biking days, "Friends don' let friends ride day-glow".

![front](https://github.com/picocomputer/discussions/assets/145504135/96913b3b-e1a8-4350-9915-d469fca13183)
![back](https://github.com/picocomputer/discussions/assets/145504135/ca8e1eb1-4a54-4ea6-9744-9f88cce9450e)

**Reply by vruumllc** [reply]
Thanks, I figured that out already. Speakers and the amp work great.

**Reply by rumbledethumps** [reply]
edit: You can run the audio validation by changing a comment in aud.c:
``` C
    int r = chan[0].nco_r; // real audio
    // int r = 0; // silence
```
It'll cycle through all the waveforms all the time until you get annoyed and reflash it. But you can test your speakers.

**Reply by rumbledethumps** [reply]
Awesome!

**Comment by rumbledethumps** [comment]
Consolized!?
![case](https://github.com/picocomputer/discussions/assets/16963588/53526ec4-f618-4d57-8c49-1c50cc3a6187)

**Reply by vruumllc** [reply]
Very nice! Great print.

It is missing the programming button extenders, though, which let you easily reprogram each pico just using your fingertip. The buttons have a little retaining ring to hold them in, and you have to drill a 6mm hole in the acrylic to make room for it, as there is not enough room for it below the acrylic. I then secure the ring to button shaft with a --tiny-- bit of superglue.

Since you have decided to go with external pico debuggers (which I think is a great call), I am working on an update that adds a 3-pin header port below each micro-usb port, with cables that will go under each pico and back to the SWD connecor. You then will be able to debug without opening the case too.

**Reply by rumbledethumps** [reply]
I drilled 2.5mm holes for a toothpick or 2mm ball driver.

**Reply by vruumllc** [reply]
That works too. Anyway, I'm glad it printed so well for you. I like the chip labels too.

**Comment by rumbledethumps** [comment]
Here's my dev kit for inspiration with your debug upgrades. I just stickied everything with dual-lock.
<details>
  <summary>Click to show</summary>

![devkit](https://github.com/picocomputer/discussions/assets/16963588/abb9893b-a0b6-4ac2-994d-f3d51d9069bc)

</details>

**Comment by vruumllc** [comment]
I added optional external debug ports to all my cases, and updated the zip file up top. 
The new debug port part will work with either the 3 pin header or the connector versions of the picos.
Note: You want to solder right angle 3-pin header on pico bottom, pointing to USB port.
![debug_port](https://github.com/picocomputer/community/assets/145504135/f70b84e6-743c-4555-a378-d94d4bf3e6b0)

**Reply by vruumllc** [reply]
Oh, I forgot to note that the new part might also be of interest to those using the 'acrylic sandwich' case.
Also, pinout looking at back of case is data (red, pin3), gnd (black, pin2), clock (yellow, pin1) in pictures.

**Comment by vruumllc** [comment]
I updated the zip files and BOM to use better video board, added info about the Li-Ion battery I used, and added a nifty new inlay part you can print to have a contrasting "RP6502" on the case front. Oh, I also added new optional battery box part.

![rp6502-w-inlay](https://github.com/picocomputer/community/assets/145504135/c3c89032-21a7-41fc-84de-6d9d9d236a71)

**Comment by rumbledethumps** [comment]
<details>
  <summary>Click to show</summary>

![smol](https://github.com/picocomputer/community/assets/16963588/e2850e63-46d8-41c4-9da6-aeda9a495cbd)

</details>

**Reply by vruumllc** [reply]
Cool! Love it. Are you going to sell them on Tindie? Check out the inlay - it adds a little pizzaz to the front.

**Reply by rumbledethumps** [reply]
That inlay does pop. The extra builds will be sent to folks who can help make the project more popular.

**Reply by vruumllc** [reply]
Excellent. Good luck. I should try to add music to Star Swarms to help the effort.

**Comment by n9jcv** [comment]
@vruumllc 
Thanks much for the design.  It is a very nice design.  I took yours and made a few mods.  The plexi is captive in the frame.  I placed it during mid print.  

I had different screws and used 4 from the back that secure the front frame.  
![20240212_233231](https://github.com/picocomputer/community/assets/6472203/864c0d7d-ce6b-461f-aaa9-8d063208215c)

**Reply by vruumllc** [reply]
Looks great! Very sleek.

**Comment by vruumllc** [comment]
I finally got around to adding these to my [github repo](https://github.com/vruumllc), and have edited the first entry to this thread accordingly

**Comment by mickstone** [comment]
Hi @vruumllc,
I'm midway through putting one of your amazing cases together and was wondering if you may be able to clarify the power setup a little?
The power pack coming in via the 2.1mm jack I'm assuming is the charger for your Li-ion pack? This is going into the buck boost converter and being boosted to I assume 12V to supply the VGA driver and panel. I wanted to make sure that I haven't misunderstood the wiring near the back of the boost converter and amp. The amp is getting 5V from the RP6502 and the RP6502 is still getting power from one of the Pi Picos micro USB? Please confirm as I'm not able to see the wiring at the back of the boost converter with clarity.

Thanks

Mick

**Reply by vruumllc** [reply]
Hi. So thrilled you are building one of my cases!

I found that the VGA driver and panel I ordered, although spec'ed for 12v, actually worked fine down to 6 volt. I therefore chose a 2C Lithium Ion battery to power the system (2 x 3.7v = 7.4v). The battery is wired directly to the 2.1 mm jack. The charger for the battery is built into the wall-wart, which supplies ~8.1v. Battery and charger are here: 

7.2V Li-Ion 9600mAH 2C battery pack, https://www.powerstream.com/un38.3-qualified-lithium-packs.htm, [$46]

Both the VGA driver board and buck boost board are also connected to the 2.1 jack through a switch, so they both get at least 7.4 volts  (from either battery or charger) when the switch is on. The buck-boost board is adjusted for a 5v output, to power the Picocomputer and the audio amp boards.

No connection to the USB ports is necessary to power the picocomputer, but it seems to be OK to use the Pico VGA USB port for the serial connection to a host computer.

![rp6502_hub_video_audio_power_wiring_photo](https://github.com/user-attachments/assets/d70c1a70-8a70-48cc-ad59-3fae61a0c164)

I hope that is all clear. If not, I can draw a diagram.

**Reply by mickstone** [reply]
Perfect, that provides exactly what I needed 💯

**Comment by ghhann** [comment]
Thank you very much for your designs.  It was exactly what is was looking for.  I actually printed up 2 both with debug ports one with the hub the other without the hub

much appreciated!

**Reply by vruumllc** [reply]
I am glad they are useful to you. Thanks for the kind words.

**Reply by ghhann** [reply]
I hate to ask but I don’t want to be a bother.  I really like the lid with the open look.  Anyway we could get a lid without the holes for the clear acrylic shield?  If it’s too much please don’t bother….  But I really enjoy the design.  All my best

**Reply by vruumllc** [reply]
Are you referring to the yellow case designed by @n9jcv ? I agree that it is a nice look.
 
I have include STEP files that you should be able to import into Autodesk Fusion, or other CAD program, and then modify as desired.
